src/util/LoadPixmapLazy.cpp

```diff
@@ -0,0 +1,50 @@
+#include "common/network/NetworkRequest.hpp"
+#include "common/network/NetworkResult.hpp"
+#include "common/QLogging.hpp"
+#include "messages/Image.hpp"
+
+#include <QBuffer>
+#include <QImageReader>
+#include <QLoggingCategory>
+#include <QPixmap>
+
+namespace chatterino {
+
+// This was copied from TwitchBadges::loadEmoteImage and modified.
+void loadPixmapFromUrlLazy(const Url &url,
+                           std::function<void(QPixmap)> &&callback)
+{
+    NetworkRequest(url.string)
+        .concurrent()
+        .cache()
+        .onSuccess(
+            [callback = std::move(callback), url](const NetworkResult &result) {
+                auto data = result.getData();
+
+                // const cast since we are only reading from it
+                QBuffer buffer(const_cast<QByteArray *>(&data));
+                buffer.open(QIODevice::ReadOnly);
+                QImageReader reader(&buffer);
+
+                if (!reader.canRead() || reader.size().isEmpty())
+                {
+                    qCWarning(chatterinoSettings)
+                        << "Can't read mod action image at" << url.string << ":"
```

Comment:
- This mentions a mod action image, but the function is more general (it can read _any_ image).
- This will add a space between the URL and the colon (same as below).

Reply:
It does add a space, to be honest I don't care that much about that space.